container/container_notsolaris.go

```diff
@@ -0,0 +1,11 @@
+// +build linux freebsd
```

Comment:
This only actually builds on Linux, FreeBSD does not have `unix.MNT_DETACH`, so looks like this should be `container_linux` and FreeBSD and Solaris can share the other file.

If you don't want to change this, feel free to leave it and I will try to tidy up the mess that is our bitrotted FreeBSD support later...